### PR TITLE
Add allneeds.lua script

### PR DESCRIPTION
**Pull request by vallode**
Taking over the work at https://github.com/DFHack/scripts/pull/90 and enacting all the comments Lethosor made.

### CODE REVIEW

File: docs/allneeds.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
+allneeds
+========
+
+.. dfhack-tool::
+    :summary: Show the cummulitative needs of all citizens.
```
Comment (myk002) [inline docs/allneeds.rst:5]:
```suggestion
    :summary: Show the cumulative needs of all citizens.
```

File: allneeds.lua [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        if fort_needs[need.id] then
+            fort_needs[need.id].cummulitative_need = fort_needs[need.id].cummulitative_need + need.need_level
+            fort_needs[need.id].cummulitative_focus = fort_needs[need.id].cummulitative_focus + need.focus_level
+            fort_needs[need.id].citizen_count = fort_needs[need.id].citizen_count + 1
+        else
+            fort_needs[need.id] = {}
+            fort_needs[need.id].cummulitative_need = need.need_level
+            fort_needs[need.id].cummulitative_focus = need.focus_level
+            fort_needs[need.id].citizen_count = 1
+        end
```
Comment (myk002) [inline allneeds.lua:21]:
```suggestion
        local needs = ensure_key(fort_needs, need.id)
        needs.cumulative_need = (needs.cumulative_need or 0) + need.need_level
        needs.cumulative_focus = (needs.cumulative_focus or 0) + need.focus_level
        needs.citizen_count = (needs.citizen_count or 0) + 1
```

File: docs/allneeds.rst (hunk 1 above)
Comment (vallode) [reply docs/allneeds.rst:5]: Wow I really went ham with the cumulitative, need more coffee

File: allneeds.lua (hunk 2 above)
Comment (myk002) [reply allneeds.lua]: this would shorten up the logic in this section

**Comment by myk002** [comment]
oh, also, since this is a new script, could you add a note to the changelog.txt file about it in the "New Scripts" section?

File: allneeds.lua (hunk 2 above)
Comment (vallode) [reply allneeds.lua]: `ensure_key` is neat, thanks :)

**Review by myk002 — APPROVED** [review]

**Pull request merged**